strawberryfields/apps/similarity.py

```diff
@@ -430,101 +559,205 @@ def prob_event_mc(
+    to be used in the estimation.
 
-    Probabilities are reconstructed by measuring the occurrence of events in the input ``samples``.
+    Computing exact probabilities for a large number of orbits especially for orbits with high
+    total photon numbers can be quite time-consuming. For example, calculating the exact
+    probabilities of observing 8 total photons in a 25 mode graph can take on the order of a
+    few minutes. This means that if the dataset has 1000 graphs, obtaining feature vectors for
+    all can take many days. Monte Carlo estimation, although less precise, can be much quicker.
 
     **Example usage:**
 
-    >>> from strawberryfields.apps import data
-    >>> samples = data.Mutag0()
-    >>> feature_vector_sampling(samples, [2, 4, 6])
-    [0.19035, 0.2047, 0.1539]
+    >>> graph = nx.complete_graph(8)
+    >>> feature_vector_orbits(graph, [ [1,1], [2], [2,1,1], [1,1,1,1] ])
+    [0.18423136914879445,
+     7.843627302119571e-31,
+     0.03744399092424445,
+     0.07020748298295838]
+
+    >>> feature_vector_orbits(graph, [ [1,1], [2] , [2,1,1], [1,1,1,1] ], mc_samples = 1000)
+    [0.1842313691487924,
+     8.098317264668192e-31,
+     0.037443990924243886,
+     0.07020748298295804]
 
     Args:
-        samples (list[list[int]]): a list of samples
-        event_photon_numbers (list[int]): a list of events described by their total photon number
-        max_count_per_mode (int): maximum number of photons per mode for all events
+        graph (nx.Graph): input graph
+        list_of_orbits (list[list[int]]): a list of orbits
+        n_mean (float): total mean photon number of the GBS device
+        mc_samples (int): number of samples used in the Monte Carlo estimation
+        loss (float): fraction of photons lost in GBS
 
     Returns:
-        list[float]: a feature vector of event probabilities in the same order as
-        ``event_photon_numbers``
+        list[float]: a feature vector of orbit probabilities
     """
-    if min(event_photon_numbers) < 0:
-        raise ValueError("Cannot request events with photon number below zero")
 
-    if max_count_per_mode < 0:
-        raise ValueError("Maximum number of photons per mode must be non-negative")
-
-    n_samples = len(samples)
+    if len(list_of_orbits) <= 0:
+        raise ValueError("At least one orbit should be provided")
+    if n_mean < 0:
+        raise ValueError("Mean photon number must be non-negative")
+    if not 0 <= loss <= 1:
+        raise ValueError("Loss parameter must take a value between zero and one")
 
-    e = (sample_to_event(s, max_count_per_mode) for s in samples)
-    count = Counter(e)
+    if mc_samples == None:
+        return [_prob_orbit_exact(graph, orbit, n_mean, loss) for orbit in list_of_orbits]
 
-    return [count[p] / n_samples for p in event_photon_numbers]
+    else:
+        return [_prob_orbit_mc(graph, orbit, n_mean, mc_samples, loss) for orbit in list_of_orbits]

```

Comment:
You could have:
```python
    if mc_samples:
        return [_prob_orbit_mc(graph, orbit, n_mean, mc_samples, loss) for orbit in list_of_orbits]

    return [_prob_orbit_exact(graph, orbit, n_mean, loss) for orbit in list_of_orbits]
```
In this way, `mc_samples=None` and `mc_samples=0` will result in exact sampling, otherwise for `samples` being an `int` it will do MC.